kubernetes/structures.go

```diff
@@ -3,15 +3,94 @@ package kubernetes
 import (
 	"encoding/base64"
 	"fmt"
-	"net/url"
 	"strings"
 
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	apps "k8s.io/api/apps/v1beta1"
+	cert "k8s.io/api/certificates/v1beta1"
 	api "k8s.io/api/core/v1"
+	discovery "k8s.io/api/discovery/v1beta1"
+	networking "k8s.io/api/networking/v1beta1"
+	rbac "k8s.io/api/rbac/v1beta1"
+
 	"k8s.io/apimachinery/pkg/api/resource"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// https://kubernetes.io/docs/reference/labels-annotations-taints
+var builtInLabels = map[string]string{
+	//Core Api
+	api.LabelHostname:                            "",
+	api.LabelZoneFailureDomain:                   "",
+	api.LabelZoneRegion:                          "",
+	api.LabelZoneFailureDomainStable:             "",
+	api.LabelZoneRegionStable:                    "",
+	api.LabelInstanceType:                        "",
+	api.LabelInstanceTypeStable:                  "",
+	api.LabelOSStable:                            "",
+	api.LabelArchStable:                          "",
+	api.LabelWindowsBuild:                        "",
+	api.LabelNamespaceSuffixKubelet:              "",
+	api.LabelNamespaceSuffixNode:                 "",
+	api.LabelNamespaceNodeRestriction:            "",
+	api.IsHeadlessService:                        "",
+	api.BetaStorageClassAnnotation:               "",
+	api.MountOptionAnnotation:                    "",
+	api.ResourceDefaultNamespacePrefix:           "",
+	api.ServiceAccountNameKey:                    "",
+	api.ServiceAccountUIDKey:                     "",
+	api.PodPresetOptOutAnnotationKey:             "",
+	api.MirrorPodAnnotationKey:                   "",
+	api.TolerationsAnnotationKey:                 "",
+	api.TaintsAnnotationKey:                      "",
+	api.SeccompPodAnnotationKey:                  "",
+	api.SeccompContainerAnnotationKeyPrefix:      "",
+	api.AppArmorBetaContainerAnnotationKeyPrefix: "",
+	api.AppArmorBetaDefaultProfileAnnotationKey:  "",
+	api.AppArmorBetaAllowedProfilesAnnotationKey: "",
+	api.PreferAvoidPodsAnnotationKey:             "",
+	api.NonConvertibleAnnotationPrefix:           "",
+	api.AnnotationLoadBalancerSourceRangesKey:    "",
+	api.EndpointsLastChangeTriggerTime:           "",
+	api.MigratedPluginsAnnotationKey:             "",
+	api.TaintNodeNotReady:                        "",
+	api.TaintNodeUnreachable:                     "",
+	api.TaintNodeUnschedulable:                   "",
+	api.TaintNodeMemoryPressure:                  "",
+	api.TaintNodeDiskPressure:                    "",
+	api.TaintNodeNetworkUnavailable:              "",
+	api.TaintNodePIDPressure:                     "",
+
+	// Secrete Types
```

Comment:
Actually I don't think we need these, because they go in the `type` field for Secret.

Reply:
removed